netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -406,11 +430,40 @@ protected ClientTransportFactory buildTransportFactory() {
       }
       negotiator = createProtocolNegotiatorByType(negotiationType, localSslContext);
     }
+
+    boolean usingSharedGroup = eventLoopGroup == null;
+
+    populateDefaultChannelAndGroup();
+
     return new NettyTransportFactory(
         negotiator, channelFactory, channelOptions,
         eventLoopGroup, flowControlWindow, maxInboundMessageSize(),
         maxHeaderListSize, keepAliveTimeNanos, keepAliveTimeoutNanos, keepAliveWithoutCalls,
-        transportTracerFactory.create(), localSocketPicker);
+        transportTracerFactory.create(), localSocketPicker, usingSharedGroup);
+  }
+
+  private void populateDefaultChannelAndGroup() {
+    boolean groupOrChannelProvided = channelFactory != DEFAULT_CHANNEL_FACTORY
+        || eventLoopGroup != null;
+
+    if (groupOrChannelProvided) {
+      logger.log(
+          Level.WARNING,
+          "Both EventLoopGroup and ChannelType should be provided, otherwise client may not start "
+              + "depends on the gRPC default value.");
+
+      if (eventLoopGroup == null) {
+        eventLoopGroup = SharedResourceHolder.get(Utils.NIO_WORKER_EVENT_LOOP_GROUP);
```

Comment:
We shouldn't be mutating the builder. That can cause quite a bit of confusion if `build()` is called twice. For example, the way this code is, the second call to build() would not acquire a reference to the event loop and so it may be shut down while still in use.